src/test/niftypet_projector/test_ProjectorNiftyPET.cxx

```diff
@@ -0,0 +1,477 @@
+    timer.stop();
+    double time_bck(timer.value());
+    std::cerr << "\tDone! (" << time_bck << " secs)\n";
+
+    time = time_fwd+time_bck;
+
+    std::cerr << "\nTotal time for projection with " << fwrd_projector.get_registered_name() << ": " << time << " secs.\n";
+
+    if (save_to_file) {
+        sino_sptr->write_to_file(name + "_forward_projected.hs");
+        shared_ptr<OutputFileFormat<DiscretisedDensity<3,float> > > output_file_format_sptr =
+                OutputFileFormat<DiscretisedDensity<3,float> >::default_sptr();
+        output_file_format_sptr->write_to_file(name + "_back_projected",*image_sptr);
+    }
+}
+
+inline bool file_exists (const std::string& name) {
+    ifstream f(name.c_str());
+    return f.good();
+}
+
+void
+TestGPUProjectors::
+set_up_input_sino()
+{
+    if (!_sinogram_filename.empty()) {
+        _proj_data_sptr = MAKE_SHARED<ProjDataInMemory>(*ProjData::read_from_file(_sinogram_filename));
+    }
+    else {
+        // Create scanner
+        shared_ptr<Scanner> scanner_sptr(new Scanner(Scanner::Siemens_mMR));
+
+        // ExamInfo
+        shared_ptr<ExamInfo> exam_info_sptr(new ExamInfo);
+        exam_info_sptr->imaging_modality = ImagingModality::PT;
+
+        // ProjDataInfo
+        int max_seg_num;
+        if (_span == 11)
+            max_seg_num = 5;
+        else if (_span == 1)
+            max_seg_num = 0;
+        else
+            throw std::runtime_error("Span of 1 and 11 only currently supported.");
+
+        VectorWithOffset<int> ax_pos_per_segment(-max_seg_num,max_seg_num);
+        ax_pos_per_segment[0] = 127;
+        if (max_seg_num > 0) {
+            ax_pos_per_segment[-1] = ax_pos_per_segment[1] = 115;
+            ax_pos_per_segment[-2] = ax_pos_per_segment[2] =  93;
+            ax_pos_per_segment[-3] = ax_pos_per_segment[3] =  71;
+            ax_pos_per_segment[-4] = ax_pos_per_segment[4] =  49;
+            ax_pos_per_segment[-5] = ax_pos_per_segment[5] =  27;
+        }
+
+        VectorWithOffset<int> max_ring_diff(-max_seg_num,max_seg_num);
+        if (max_seg_num > 0) {
+            max_ring_diff[-5] = -50;
+            max_ring_diff[-4] = -39;
+            max_ring_diff[-3] = -28;
+            max_ring_diff[-2] = -17;
+            max_ring_diff[-1] = -06;
+        }
+        max_ring_diff[ 0] =  05;
+        if (max_seg_num > 0) {
+            max_ring_diff[ 1] =  16;
+            max_ring_diff[ 2] =  27;
+            max_ring_diff[ 3] =  38;
+            max_ring_diff[ 4] =  49;
+            max_ring_diff[ 5] =  60;
+        }
+
+        VectorWithOffset<int> min_ring_diff(-max_seg_num,max_seg_num);
+        for (int i=-max_seg_num; i<=max_seg_num; ++i)
+            min_ring_diff[i] = -max_ring_diff[-i];
+
+        shared_ptr<ProjDataInfoCylindricalNoArcCorr> proj_data_info_sptr
+                (new ProjDataInfoCylindricalNoArcCorr
+                 (scanner_sptr,ax_pos_per_segment,min_ring_diff,max_ring_diff,scanner_sptr->get_max_num_views(),344));
```

Comment:
All of this should be a call to 
```
  static unique_ptr<ProjDataInfo>
	  ProjDataInfo::construct_proj_data_info(const shared_ptr<Scanner>& scanner_sptr,
		  const int span, const int max_delta,
		  const int num_views, const int num_tangential_poss,
		  /* arc_corrected */ false)
```

Reply:
What should I use for `max_delta`? 

I tried with `scanner_sptr->get_num_rings()-1`, which I'd seen elsewhere, but this results in the wrong segment range (`(-6,6)` instead of `(5,5)`).

Reply:
for the mMR it's 60 (for somewhat complicated reasons)

Reply:
ok